src/lib/state/reducers/query.js

```diff
@@ -92,7 +92,8 @@ export default (state = {}, action) => {
     case SET_QUERY_STATE:
       return {
         ...state,
-        ...updateQueryState(state, action.payload, INITIAL_QUERY_STATE_KEYS),
+        ...INITIAL_QUERY_STATE,
```

Comment:
As discussed, we should keep it to avoid that extra keys, coming externally (backend response), will be injected in the query state.

I would also rename the method `updateQueryState` to something more clear, e.g. `sanitize...`